cpp/include/rapidsmpf/buffer/resource.hpp

```diff
@@ -155,16 +155,19 @@ class BufferResource {
      * continuously checks and perform spilling based on the memory availability
      * functions. The value of `periodic_spill_check` is used as the pause between checks.
      * If `std::nullopt`, no periodic spill check is performed.
+     * @param stream_pool_size The number of streams to create in the CUDA stream pool.
+     * Only used if `stream_pool` is not provided.
```

Comment:
This is a weird interface. Why not just accept a stream pool and let the user construct the stream pool if they want a custom size?

Reply:
Yay! Do it!

Reply:
I think https://github.com/rapidsai/rapidsmpf/pull/633/files#diff-049939465cbe3b3d4a7593bac281eecc840a6c0cfcf78042ae23b6389859190bR108 is the gymnastics you were referring to.